### PR TITLE
foundationdb: 7.1.32 -> 7.3.42

### PR DESCRIPTION
**Pull request by emilazy**
Includes fixes for GCC 14 and using non‐vendored libraries. The alternative to bumping to the next major version here would be to always build with Clang, which might not be a terrible idea anyway as it’s what upstream use for their official binaries on all platforms (or to patch 7.1 for the missing `#include` and potentially other issues, but it seemed like we were overdue for an update anyway and there’s a clean upgrade path).

Everything builds and the NixOS Stalwart Mail test passes, though I don’t know if that exercises FoundationDB or not. cc @happysalada @onny @oddlama @Pandapip1 as Stalwart Mail maintainers.

cc @thoughtpolice – are you still using/maintaining this? Before this the package hadn’t seen an update since May 2023, which seems a bit worrying for database software (though I guess FoundationDB is probably more stable than most). It seems like the only in‐tree user is Stalwart Mail.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>foundationdb</li>
    <li>foundationdb.debug</li>
    <li>foundationdb.dev</li>
    <li>foundationdb.lib</li>
    <li>foundationdb.pythonsrc</li>
    <li>python312Packages.foundationdb73</li>
    <li>python312Packages.foundationdb73.dist</li>
    <li>python313Packages.foundationdb73</li>
    <li>python313Packages.foundationdb73.dist</li>
    <li>stalwart-mail</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4903,7 +4903,7 @@ self: super: with self; {
 
   formulaic = callPackage ../development/python-modules/formulaic { };
 
-  foundationdb71 = callPackage ../servers/foundationdb/python.nix { foundationdb = pkgs.foundationdb71; };
```
Comment (Pandapip1) [inline pkgs/top-level/python-packages.nix:4906]: This seems to be missing a throw alias?

**Review by Pandapip1 — APPROVED** [review]
Other than the one not, diff LGTM.

**Comment by happysalada** [comment]
Foundationd tags latest as 7.3.43 and everything after as prerelease. Ive been using 7.3.43 at day job but there have been some problems with it (making a multi region deployment) that dont happen with 7.1
All in all im not against updating to 7.3, but their official documentation still defaults to 7.1
7.3 is not the latest stable i think.

Personally i would use 7.3.43 rather than 56. I dont feel too strongly about that though, happy to merge the PR as is. We should just be aware that there might still be rough spots with 7.3 that are not in 7.1

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (emilazy) [reply pkgs/top-level/python-packages.nix:4906]: https://github.com/NixOS/nixpkgs/pull/369145/files#diff-4878bf3fa499d7d0bab51ebe52427ba7ee884458535a7bbae5e31f245994dbf8

**Comment by emilazy** [comment]
Oops, yes, I even saw that some of the versions were pre‐releases but just forgot when deciding to try 7.3. I’ll put it at the actual latest stable 7.3. I believe that 7.3.43 is considered stable and production‐ready per https://github.com/apple/foundationdb?tab=readme-ov-file#getting-started, and 7.1 is in maintenance mode (until 7.5, presumably). https://apple.github.io/foundationdb/ seems to point to 7.3.57; do you have a pointer to where they’re recommending 7.1 as a default?

I’m not opposed to sticking on 7.1 either but we’ll either have to match their use of Clang (to hide the error) or effectively maintain compatibility for GCC 14 ourselves (which might not be too bad if the churn is low enough). It seemed like getting up to date early in the 25.05 cycle and dropping some of our patches would be preferable to doing that work, but as I’m not a FoundationDB user myself I don’t know how much churn upgrades cause.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (Pandapip1) [reply pkgs/top-level/python-packages.nix:4906]: Huh, must've missed that. Feel free to mark this as resolved.

**Comment by happysalada** [comment]
Changing to 7.3 sounds good to me.

Last time i checked the docs , the default was 7.1, they must have changed it since then.
Very weird that the doc shows 7.3.57 even though they say its a pre release.
7.3.43 is fhe safe choice i feel but i dont feel too strongly about it.

**Comment by emilazy** [comment]
Adjusted to 7.3.42 (not .43, as that’s the AVX version – why do they do it like that??); Stalwart continues to build.

@happysalada Are you interested in maintaining this package, by the way? Seems like it could use some attention for version bumps.

**Comment by happysalada** [comment]
Thank you.
I think i read on the roadmap that there is a big update planned for march 2025.
Ill try to make the update at that time and add myself as a maintainer. I cant comlmit right now though, but im hoping the situation will get better at day job until then .

**Comment by happysalada** [comment]
This PR looks good to me, not sure why ofborg is not evaluating.

**Comment by emilazy** [comment]
ofborg eval is very slow these days. The GitHub Actions evaluation passed.

**Comment by JulienMalka** [comment]
I have a build failure for foundationdb 7.3.42, as is also reported here: https://github.com/NixOS/nixpkgs/issues/370345

**Pull request merged**